MyModels.py

```diff
@@ -0,0 +1,158 @@
+class BaseModel:
+    def __init__(self):
+        self.list = []
+        self.id_index = {}
+        
+    def add(self, entry):
+        
+        self.list.append(entry)
+    
+#     def find_index(self,entry):
+#         for i, item in self.list:
+#             if item==entry:
+#                 return i
+            
+#     def find_val(self,index):
+#         return self.list[index]
+
+#####################
+
+    def find_index(self,id):
+        return self.id_index[id]
+    
+    def find_val(self,id):
+        index = self.find_index(id)
+        return self.list[index]
+
+
+###################
+    def GetAll(self):
+        return self.list
+    
+    def delete(self, entry):
+        
+        self.list.remove(entry)
+    
+    def update(self,id, new_entry):
+        
+        index = self.id_index[id]
+        self.list[index] = new_entry
+        
+    
+
+class User(BaseModel):
+    
+    def __init__(self):
+        self.email_userid = {}
+        self.userid = 0
+        super().__init__()
+        
+    
+    def register(self, client_name, email, password):
+        entry = {'client_name' : client_name,
+                'email': email,
+                'password':password,
+                'userid': self.userid }
+        self.add(entry = entry)
+        self.id_index[self.userid] = len(self.list) - 1
+        self.email_userid[email] = self.userid
+        
+        self.userid += 1
```

Comment:
You were supposed to rename this to `id_sequence_counter`, or `id_seq_counter` if you prefer shorter names. Also this action is common between all models and needs to go up into the `BaseModel`.

Reply:
Now, a question arises here, I will ask it in Telegram.